Source/AliveLib/Glukkon.cpp

```diff
@@ -119,6 +129,130 @@ TintEntry stru_5546B4[18] =
     { 0u, 0u, 0u, 0u }
 };
 
+signed int CC Glukkon::CreateFromSaveState_442830(const BYTE* pData)
+{
+    const Glukkon_SaveState* pSaveState = reinterpret_cast< const Glukkon_SaveState* >( pData );
+    auto pTlv = static_cast< Path_Glukkon* >( sPath_dword_BB47C0->TLV_From_Offset_Lvl_Cam_4DB770(pSaveState->field_44_tlvInfo) );
+
+    switch (gMap_5C3030.sCurrentLevelId_5C3030)
+    {
+        case LevelIds::eFeeCoDepot_5:
+        case LevelIds::eFeeCoDepot_Ender_12:
+            if (!ResourceManager::GetLoadedResource_49C2A0(1835626049, 802, 0, 0))
+            {
+                ResourceManager::LoadResourceFile_49C170("ASLIK.BND", 0);
+            }
+            break;
+        case LevelIds::eBarracks_6:
+        case LevelIds::eBarracks_Ender_13:
+            if (!ResourceManager::GetLoadedResource_49C2A0(1835626049, 804, 0, 0))
+            {
+                ResourceManager::LoadResourceFile_49C170("DRIPIK.BND", 0);
+            }
+            break;
+        case LevelIds::eBonewerkz_8:
+        case LevelIds::eBonewerkz_Ender_14:
+            if (!ResourceManager::GetLoadedResource_49C2A0(1835626049, 806, 0, 0))
+            {
+                ResourceManager::LoadResourceFile_49C170("PHLEG.BND", 0);
+            }
+            break;
+        default:
+            if (!ResourceManager::GetLoadedResource_49C2A0(1835626049, 800, 0, 0))
+            {
+                ResourceManager::LoadResourceFile_49C170("GLUKKON.BND", 0);
+            }
+            break;
+    }
+
+    auto pGlukkon = alive_new<Glukkon>();
+    if (pGlukkon)
+    {
+        pGlukkon->ctor_43F030(pTlv, pSaveState->field_44_tlvInfo);
+    }
+    else
+    {
+        pGlukkon = 0;
+    }
+    pGlukkon->field_4_typeId = pSaveState->field_8E;
+    pGlukkon->field_C_objectId = pSaveState->field_4_object_id;
+    if (pSaveState->field_40_bIsActiveChar)
+    {
+        sControlledCharacter_5C1B8C = pGlukkon;
+    }
+
+    pGlukkon->field_FC_pPathTLV = nullptr;
+    pGlukkon->field_100_pCollisionLine = nullptr;
+
+    pGlukkon->field_B8_xpos = pSaveState->field_8_xpos;
+    pGlukkon->field_BC_ypos = pSaveState->field_C_ypos;
+
+    pGlukkon->field_C4_velx = pSaveState->field_10_xvel;
+    pGlukkon->field_C8_vely = pSaveState->field_14_yvel;
+
+    pGlukkon->field_1D8 = pSaveState->field_58;
+    pGlukkon->field_C0_path_number = pSaveState->field_18_path;
+    pGlukkon->field_C2_lvl_number = pSaveState->field_1A_level;
+    pGlukkon->field_CC_sprite_scale = pSaveState->field_1C_sprite_scale;
+
+    pGlukkon->field_D0_r = pSaveState->field_20_r;
+    pGlukkon->field_D2_g = pSaveState->field_22_g;
+    pGlukkon->field_D4_b = pSaveState->field_24_b;
+
+    pGlukkon->field_1A0 = pSaveState->field_20_r; //todo delete if nothing uses it

```

Comment:
isn't it used in vRender?

Reply:
Possibly, but vRender isn't done yet. I'll handle it, anyway.